services/py-api/src/database/db_manager.py

```diff
@@ -66,6 +67,35 @@ def get_collection(self, collection_name: str) -> AsyncIOMotorCollection:
         # https://pymongo.readthedocs.io/en/stable/tutorial.html#getting-a-collection
         return self._client[self._DB_NAME][collection_name]
 
+    @staticmethod
+    async def retry_db_operation[T](db_operation: Callable[..., Awaitable[T]], *args: Any, **kwargs: Any) -> T:
+        """
+        If there was a TransientTransactionError during execution, retries the db operation with exponential backoff up
+        to max_retries times. Having an exponential backoff increases our chances of success if there was a temporary
+        network issue.
+
+        Raises:
+            PyMongoError
+        """
+        max_retries = 3
+        delay = 1  # initial delay in seconds
```

Comment:
Isn't this too big of a delay? Especially, if you're doing an exponential backoff this seems to be a pretty high value.

Reply:
fixed in 43821673891e234b5d62551c4ee093940a082253 and c6a67ae107d5b1a2fc4edfcb3725db905d2acdce